api.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"github.com/gin-gonic/gin"
+	"log"
+	"net/http"
+)
+
+/*
+ * Returns a HelloWorld json message
+ */
+func helloHandler(c *gin.Context) {
+	c.JSON(http.StatusOK, gin.H{"message": "Hello World!"})
+}
+
+/*
+ * Returns a greeting json message
+ */
+func greetHandler(c *gin.Context) {
+	name := c.Param("name")
+	c.JSON(http.StatusOK, gin.H{"greet-message": "Hello " + name + "!"})
+}
+
+/*
+ * Returns a json array of all the products
+ */
+func GetAllPatients(c *gin.Context) {
+
+	rows, err := db.Query("SELECT * FROM patient")
+	if err != nil {
+		panic(err)
+	}
+	defer rows.Close()
+
+	var patients []patient
+	for rows.Next() {
+		row := patient{}
+		err = rows.Scan(&row.ID, &row.Name, &row.Illness, &row.BirthDate, &row.LocationId)
+
+		if err != nil {
+			panic(err)
+		}
+		patients = append(patients, row)
+	}
+
+	err = rows.Err()
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	c.JSON(http.StatusOK, gin.H{"patients": patients})
+}
+
+/*
+ * Returns a json array of all the locations
+ */
+func GetAllLocations(c *gin.Context) {
+	rows, err := db.Query("SELECT * FROM location")
+	if err != nil {
+		panic(err)
+	}
+	defer rows.Close()
+
+	var locations []location
+	for rows.Next() {
+		row := location{}
+		err = rows.Scan(&row.ID, &row.Name, &row.HospitalId)
+
+		if err != nil {
+			panic(err)
+		}
+		locations = append(locations, row)
+	}
+
+	err = rows.Err()
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	c.JSON(http.StatusOK, gin.H{"locations": locations})
+}
+
+/*
+ * Returns json array of all the hospitals
+ */
+func GetAllHospitals(c *gin.Context) {
+	rows, err := db.Query("SELECT * FROM hospital")
+	if err != nil {
+		panic(err)
+	}
+	defer rows.Close()
+
+	var hospitals []hospital
+	for rows.Next() {
+		row := hospital{}
+		err = rows.Scan(&row.ID, &row.Name, &row.HospitalId)
+
+		if err != nil {
+			panic(err)
+		}
+		hospitals = append(hospitals, row)
+	}
+
+	err = rows.Err()
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	c.JSON(http.StatusOK, gin.H{"hospitals": hospitals})
```

Comment:
No need to wrap with `"hospitals"` attribute, just return the `hospitals` array.